[pytorch] Fix printing of optional string arguments in schemas

SplitInfinity opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#55196 [pytorch] Fix printing of optional string arguments in schemas**

This commit fixes printing of default values for optional string type arguments in schemas. At the moment, these default values are not printed as quoted strings. If a schema with an optional string type parameter with a default value that is not `None` is printed and then parsed, the lack of quotes causes a parsing error.

Differential Revision: [D27525450](https://our.internmc.facebook.com/intern/diff/D27525450/)

Commit:
[pytorch] Fix printing of optional string arguments in schemas

This commit fixes printing of default values for optional string type arguments in schemas. At the moment, these default values are not printed as quoted strings. If a schema with an optional string type parameter with a default value that is not `None` is printed and then parsed, the lack of quotes causes a parsing error.

Differential Revision: [D27525450](https://our.internmc.facebook.com/intern/diff/D27525450/)

[ghstack-poisoned]

gmagogsfm commented on aten/src/ATen/core/function_schema.h:
```diff
@@ -406,7 +406,7 @@ inline std::ostream& operator<<(std::ostream& out, const Argument& arg) {
   if (arg.default_value()) {
     out << "=";
-    if (type->kind() == c10::TypeKind::StringType) {
+    if (type->kind() == c10::TypeKind::StringType || unopt_type->kind() == c10::TypeKind::StringType) {
```
The change itself looks good.

Just reminds me of current `Union` work by @ansley. Looks like this logic needs to be updated again after `Union` lands?

Commit:
Update on "[pytorch] Fix printing of optional string arguments in schemas"


This commit fixes printing of default values for optional string type arguments in schemas. At the moment, these default values are not printed as quoted strings. If a schema with an optional string type parameter with a default value that is not `None` is printed and then parsed, the lack of quotes causes a parsing error.

Differential Revision: [D27525450](https://our.internmc.facebook.com/intern/diff/D27525450/)

[ghstack-poisoned]
